### DISCUSSION TITLE
Can i use TypeScript？

### DISCUSSION THREAD

**Discussion by hotuns** [MelihAltintas/vue3-openlayers#45] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

openlayer 已经有 @types/ol ,但是 vue3-openlayers不能使用它

现在vue3-openlayers内是通过 provide ， inject来获取。但是这样没有类型了
    app.provide('ol-feature',feature)
    app.provide('ol-geom',geom)
    app.provide('ol-animations',animations)
    app.provide('ol-format',format)
    app.provide('ol-loadingstrategy',loadingstrategy)
    app.provide('ol-selectconditions',selectconditions)
    app.provide('ol-extent',extent)

我要怎么做才能在const selectConditions = inject('ol-selectconditions')   获取到selectConditions的类型呢

**Describe alternatives you've considered**

**Additional context**

**Comment by MelihAltintas** [comment, marked as answer]
Yes BUT d.ts files is not written without type checking you can use this lib like that:

In shims-vue.d.ts you should add this line:
`declare module 'vue3-openlayers`

in main.ts :
```
import { createApp } from 'vue'
import App from './App.vue'

import OpenLayersMap from 'vue3-openlayers'
import 'vue3-openlayers/dist/vue3-openlayers.css'

const app = createApp(App);
app.use(OpenLayersMap)

app.mount('#app')
```

**Reply by BernhardSchlegel** [reply]
Can you elaborate how this is working?

**Reply by MelihAltintas** [reply]
@BernhardSchlegel 
 https://github.com/gn01792218/f2e-tourism this project is typescript based. You can view shims-vue.d.ts in src folder. You should add one line declare module vue3-openlayers
